src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngineConfig.java

```diff
@@ -65,4 +82,21 @@ public Boolean shouldRetainTables() {
   public Integer getTempTableTTLHours() {
     return tempTableTTLHours != null && tempTableTTLHours > 0 ? tempTableTTLHours : 72;
   }
+
+  public QueryJobConfiguration.Priority getJobPriority() {
+    String priority = jobPriority != null ? jobPriority : "batch";
+    return QueryJobConfiguration.Priority.valueOf(priority.toUpperCase());
+  }
+
+  /**
+   * Validates configuration properties
+   */
+  public void validate() {
+    // Ensure value for the job priority configuration property is valid
+    if (jobPriority != null && !containsMacro(NAME_JOB_PRIORITY)
+      && !PRIORITY_BATCH.equalsIgnoreCase(jobPriority)
+      && !PRIORITY_INTERACTIVE.equalsIgnoreCase(jobPriority)) {
+      throw new SQLEngineException("Property 'jobPriority' must be 'batch' or 'interactive'");
```

Comment:
Unfortunately, the Failure Collector is not available in the context of the SQL Engine execution, so I have to throw an exception here.

Reply:
would be good to open a jira to add this capability in some later release. I imagine it would be useful to have a 'validate' button for the sql engine.